src/prefect/orion/models/block_schemas.py

```diff
@@ -49,9 +49,18 @@ async def create_block_schema(
         exclude={"block_type", "id", "created", "updated"},
     )
     definitions = definitions or insert_values["fields"].pop("definitions", None)
-    insert_values["checksum"] = Block._calculate_schema_checksum(
-        insert_values["fields"]
+    checksum = Block._calculate_schema_checksum(insert_values["fields"])
```

Comment:
My brain is hurting a bit trying to think through the implications here, especially with nested schemas.
Is the server the source of truth for creating all nested checksums as well? 🤔

Reply:
Yes, the server is the source of truth for all nested block checksums. The client will send a block schema with fields that have block schema checksums for block schema references, but the server is the one that ultimately determines what checksum is stored for a block schema. The checksums for block schema references sent by the client are used by the server to determine if a nested block schema is already registered, but ultimately the server saves the checksum that it computes for each block schema.